macros/src/lib.rs

```diff
@@ -255,6 +258,23 @@ fn build_op_builder(
         }
     }
 
+    if return_type.is_some() {
+        builder_fields.push(quote! {
+            return_type: Option<Type>,
+        });
+
+        builder_setters.push(quote! {
+            return_type: None,
+        });
+
+        builder_accessors.push(quote! {
+            pub fn return_type(mut self, ty: Type) -> Self {
```

Comment:
FIXME: it should not nate `Type`, but what user provided (like `#return_type` or smth)